src/mapping/Mapping.cpp

```diff
@@ -9,16 +9,18 @@ namespace precice {
 namespace mapping {
 
 Mapping::Mapping(
-    Constraint constraint,
-    int        dimensions,
-    bool       requireGradient)
+    Constraint            constraint,
+    int                   dimensions,
+    bool                  requireGradient,
+    Mapping::CouplingKind couplingKind)
     : _requireGradient(requireGradient),
       _constraint(constraint),
       _inputRequirement(MeshRequirement::UNDEFINED),
       _outputRequirement(MeshRequirement::UNDEFINED),
       _input(),
       _output(),
-      _dimensions(dimensions)
+      _dimensions(dimensions),
```

Comment:
Yes, seems like we have to re-order these things in order to get rid of the warnings here